### PR TITLE
network explorer - message graph

### PR DESCRIPTION
**Pull request by rjwebb**
Messages in Canvas are partially ordered. This PR updates the network explorer user interface to display a tree view to the left of the list of messages. The tree is an SVG rendered using the `<canvas>` tag. By default if you click through to the topic page from the homepage you will see the table view, then you can click "Network view" to view the network graph. We don't have a button library so I just wrote something minimal with tailwind classes.

The network API endpoint will probably be slow if we have a huge number of messages - I haven't managed to test this with a large workload, we may want to paginate or incrementally load this data.

There are some changes to core and gossiplog in this PR in order to expose the "branch" field for the messages.

Screenshot 2024-09-12 at 11 12 19 AM

### CODE REVIEW

File: packages/gossiplog/src/AbstractGossipLog.ts [hunk 1]
```diff
@@ -189,7 +189,7 @@ export abstract class AbstractGossipLog<Payload = unknown> extends TypedEventEmi
 		const { reverse = false, limit, ...where } = range
 		return this.db.query<{ id: string; signature: Signature; message: Message<Payload> }>("$messages", {
 			where: { id: where },
-			select: { id: true, signature: true, message: true },
```
Comment (raykyri) [inline packages/gossiplog/src/AbstractGossipLog.ts:192]: Should we add branch to the return type above?

File: packages/gossiplog/src/AbstractGossipLog.ts [hunk 2]
```diff
@@ -174,12 +174,12 @@ export abstract class AbstractGossipLog<Payload = unknown> extends TypedEventEmi
 		return records.length > 0
 	}
 
-	public async get(id: string): Promise<[signature: Signature, message: Message<Payload>] | [null, null]> {
+	public async get(id: string) {
 		const record = await this.db.get<MessageRecord<Payload>>("$messages", id)
 		if (record === null) {
-			return [null, null]
+			return null
```
Comment (raykyri) [inline packages/gossiplog/src/AbstractGossipLog.ts:180]: Why change the get API here - is there a place where destructuring wouldn't work? It is a breaking change for downstream users and the destructuring based API seems okay to me upon a first look, so just wondering about the rationale for that.

**Comment by raykyri** [comment]
We'll have to change how the API fetching works in the future and probably just support message graphs for sections of the causal log, but this looks good for the current version.

File: packages/gossiplog/src/AbstractGossipLog.ts (hunk 2 above)
Comment (rjwebb) [reply packages/gossiplog/src/AbstractGossipLog.ts:180]: The goal here (see line 182) is to return `branch` in the response to `get`. Should I instead return something like `[signature, message, branch]`?

File: packages/gossiplog/src/AbstractGossipLog.ts (hunk 1 above)
Comment (rjwebb) [reply packages/gossiplog/src/AbstractGossipLog.ts:192]: Yes, good catch

**Pull request merged**